proposals/csharp-12.0/collection-expressions.md

```diff
@@ -219,7 +213,36 @@ ImmutableArray<int> ia =
 ## Construction
 [construction]: #construction
 
-*Give specific ordering for determining how to construct the constructible collection types.*
+A collection expression has a *known length* if the compile-time type of each *spread element* in the collection expression is [*countable*](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/ranges.md#adding-index-and-range-support-to-existing-library-types).
+
+*Does *countable* include explicit implementations of `ICollection<T>.Count` and `ICollection.Count`? We should check for those interfaces as well.*
```

Comment:
Precedent from list patterns is to ignore the count if it's only provided in an explicit implementation

https://sharplab.io/#v2:EYLgtghglgdgNAFxFANnAJiA1AHwAIBMAjALABQeADAAR5EB0AwgPYooCmAxglMzAM4BuclVpEALMLLkAbhABO1TtQC81GOwDu1RgAoAlFKgAzaruVR+1ANpE4BALr7qAb2oBfciIDMtAjuoQagBJFjYuHj4AHlgEAD5yF3JqFOpYkLCObl4YGJh4pmYAV3zVOOoEAAt5Zm0NbQA5ZgRgsAAHDjB2fPZ0AFEAD052NsiYAylk1OBmVgzWLLG8guD+ACV2CHQAeRgUAE8yiura9S1qJpb2zu6EXsHh0ZyJrzJU2nF58Ozo2Lj6ACC6HQunSUDuYH0UxSSTe71SVRqdXOl1aHXYXR6/SGIzGLzhKU80gJHy+ixyy3+jA4CgM0Nc9PeiNO9QuzTRNyxD1xz0M9KJ9Jmc1CCwiFL+hXy0AEoNK4IxUJJsPhCJOyMa7OuGNu9xxTz4+PeApJeE+Iu+SwlLDa+wAKsxZQhrA5qAp5BB9nA0qU3R7gjB0OwBor3sqVcckWcNVd0Zi7tjHni+STje9gn0YEUuu6EMx5JSxN5KfQAOLsBAZrPsHN5ulKxmqyOs1FauO6xO8qRG15pyvZiC5xR0byl8t96sD2sh1JhlXM9VsmOc+Pc/XjZPd4nvIUoMli375f4bMDMGTsR1pCHTmENlLzqOLjnarl6pNd1JE9xAA==